src/enclave/enclave_oe.c

```diff
@@ -294,9 +294,20 @@ void sgxlkl_ethread_init(void)
     /* Initialization completed, now run the scheduler */
     init_ethread_tp();
     _lthread_sched_init(sgxlkl_enclave_state.config->stacksize);
-    lthread_run();
 
-    return;
+    int terminating_ethread = lthread_run();
+
+    // This may be the terminating ethread exiting, so return the exit status
+    // from the enclave.
+    if (terminating_ethread)
+    {
+        return sgxlkl_enclave_state.exit_status;
+    }
+    else
+    {
+        // This ethread is not the terminating ethread
+        return INT_MAX;
+    }
```

Comment:
This code seems to be duplicated in both callers of `lthread_run`, can we move it into `lthread_run`?

Reply:
Done.